cron_service/src/modules/database/subscriptions/cron-subscriptions/cron-subscriptions.service.ts

```diff
@@ -0,0 +1,119 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { Mutex } from 'async-mutex';
+import { parseExpression } from 'cron-parser';
+import { Model } from 'mongoose';
+import { CronConnector } from '../../../connectors/cron.connector';
+import {
+  CronSubscription,
+  CronSubscriptionsDocument,
+} from './cron-subscriptions.model';
+
+@Injectable()
+export class CronSubscriptionsService {
+  private logger = new Logger(CronSubscriptionsService.name);
+  private cronSubscriptionsCache: CronSubscriptionsDocument[] = [];
+  private cacheMutex = new Mutex();
```

Comment:
We should add a 10 seconds timeout. If the worker waits for more than 10 seconds, then it should give up and let the next cron trigger try.

```ts
const mutexWithTimeout = withTimeout(new Mutex(), 100, new Error('new fancy error'));
```